src/recovery/errors.rs

```diff
@@ -14,6 +14,9 @@ pub enum RecoveryError {
     #[error("failed to fetch release data from server")]
     ApiError(#[from] ApiError),
 
+    #[error("generic error")]
+    Anyhow(#[from] anyhow::Error),
```

Comment:
`anyhow` should really only be used in *binary* code, not in *library* code. Anyhow is a convenience for errors whose destiny is to be reported, rather than handled. A *library* shouldn't be making decisions about *how* it's errors are handling (that's up to the application). The library should use concrete error types (thiserror).

This is slightly more murky in this project because of the shared source directory pattern being used (lib + bin in one source directory), but it's rarely a good idea for a single build target to mix both `thiserror` and `anyhow`.